### PR TITLE
fix widget height / vertical position issues

### LINKED ISSUE #207601: Resizing terminal will not resize the terminal chat widget
Repro:

1. Open terminal chat
2. Ask "list files in various ways"
3. Reduce height of terminal

image

### LINKED ISSUE #207602: Terminal chat widget with a lot of content cannot be scrolled
Repro:

1. Open terminal chat
2. Ask "list files in various ways", 🐛 content is clipped at bottom but scrolling doesn't work

image

### PR DESCRIPTION
**Pull request by meganrogge**
fixes #207602
fixes #207601

### CODE REVIEW

File: src/vs/workbench/contrib/inlineChat/browser/inlineChatWidget.ts [hunk 1]
```diff
@@ -312,7 +312,7 @@ export class InlineChatWidget {
 	protected _doLayout(widgetDimension: Dimension, inputDimension: Dimension): void {
 		this._elements.progress.style.width = `${inputDimension.width}px`;
 		this._chatMessageContents.style.width = `${widgetDimension.width - 10}px`;
-		this._chatMessageContents.style.maxHeight = `270px`;
+		this._chatMessageContents.style.maxHeight = Math.min(270, widgetDimension.height) + 'px';
```
Comment (Tyriar) [inline src/vs/workbench/contrib/inlineChat/browser/inlineChatWidget.ts:315]: We'll want to make sure this is handled well in the editor, try getting it to increase and reduce size to test

**Review by Tyriar — APPROVED** [review]

File: src/vs/workbench/contrib/inlineChat/browser/inlineChatWidget.ts (hunk 1 above)
Comment (jrieken) [reply src/vs/workbench/contrib/inlineChat/browser/inlineChatWidget.ts:315]: What problem does this fix? I don't believe setting overall height as messages height is correct.

Comment (Tyriar) [reply src/vs/workbench/contrib/inlineChat/browser/inlineChatWidget.ts:315]: The PR fixed the widget getting clipped when resized as it's in a fixed position on top of the terminal https://github.com/microsoft/vscode/issues/207601

Comment (jrieken) [reply src/vs/workbench/contrib/inlineChat/browser/inlineChatWidget.ts:315]: I believe there are bugs and this change might fix something but I don't believe this is the right fix. The max height of the chat contents cannot be the overall widget height.

**Pull request merged**